File: C/swap_function.c
#include <stdio.h>

void swap(int x ,int y) {
    int temp = x;
    x = y;
    y = temp;
}

int main() {
    int x = 10;
    int y = 5;

    swap(x, y);
    // variable nya beda linkungan boss, ya gak bisa di swaplah
    printf("ini x = %d\n", x);
    printf("ini y = %d\n", y);
    return 0;
}

File: go/swap_function.go
package main
import "fmt"

func swap(x int, y int) {
	var temp int = x
    x = y
    y = temp
}

func main() {
	var x int = 1000
	var y int = 20

	// sama juga gak bisa
	swap(x, y)

	fmt.Println(x)
	fmt.Println(y)

}